src/Tribe/REST/V1/Main.php

```diff
@@ -0,0 +1,191 @@
+
+	/**
+	 * Event Tickets REST API URL prefix.
+	 *
+	 * This prefx is appended to the Modern Tribe REST API URL ones.
+	 *
+	 * @var string
+	 */
+	protected $url_prefix = '/tickets/v1';
+
+	/**
+	 * @var array
+	 */
+	protected $registered_endpoints = array();
+
+	/**
+	 * Binds the implementations needed to support the REST API.
+	 *
+	 * @since TBD
+	 *
+	 */
+	public function bind_implementations() {
+		tribe_singleton( 'tickets.rest-v1.messages', 'Tribe__Tickets__REST__V1__Messages' );
+		tribe_singleton( 'tickets.rest-v1.headers-base', 'Tribe__Tickets__REST__V1__Headers__Base' );
+		tribe_singleton( 'tickets.rest-v1.settings', 'Tribe__Tickets__REST__V1__Settings' );
+		tribe_singleton( 'tickets.rest-v1.system', 'Tribe__Tickets__REST__V1__System' );
+		tribe_singleton( 'tickets.rest-v1.validator', 'Tribe__Tickets__REST__V1__Validator__Base' );
+		tribe_singleton( 'tickets.rest-v1.repository', 'Tribe__Tickets__REST__V1__Post_Repository' );
+
+		include_once Tribe__Tickets__Main::instance()->plugin_path . 'src/functions/advanced-functions/rest-v1.php';
+	}
+
+	/**
+	 * Hooks the filters and actions required for the REST API support to kick in.
+	 *
+	 * @since TBD
+	 *
+	 */
+	public function hook() {
+		$this->hook_headers();
+		$this->hook_settings();
+
+		/** @var Tribe__Tickets__REST__V1__System $system */
+		$system = tribe( 'tickets.rest-v1.system' );
+
+		if ( ! $system->supports_et_rest_api() || ! $system->et_rest_api_is_enabled() ) {
+			return;
+		}
+
+		add_action( 'rest_api_init', array( $this, 'register_endpoints' ) );
+
+	}
+
+	/**
+	 * Hooks the additional headers and meta tags related to the REST API.
+	 *
+	 * @since TBD
+	 *
+	 */
+	protected function hook_headers() {
+		/** @var Tribe__Tickets__REST__V1__System $system */
+		$system = tribe( 'tickets.rest-v1.system' );
+		/** @var Tribe__REST__Headers__Base_Interface $headers_base */
+		$headers_base = tribe( 'tickets.rest-v1.headers-base' );
+
+		if ( ! $system->et_rest_api_is_enabled() ) {
+			if ( ! $system->supports_et_rest_api() ) {
+				tribe_singleton( 'tickets.rest-v1.headers', new Tribe__REST__Headers__Unsupported( $headers_base, $this ) );
+			} else {
+				tribe_singleton( 'tickets.rest-v1.headers', new Tribe__REST__Headers__Disabled( $headers_base ) );
+			}
+		} else {
+			tribe_singleton( 'tickets.rest-v1.headers', new Tribe__REST__Headers__Supported( $headers_base, $this ) );
+		}
+
+		/** @var Tribe__REST__Headers__Headers_Interface $headers */
+		$headers = tribe( 'tickets.rest-v1.headers' );
+
+		add_action( 'wp_head', array( $headers, 'add_header' ), 10, 0 );
+		add_action( 'template_redirect', array( $headers, 'send_header' ), 11, 0 );
+	}
+
+	/**
+	 * Hooks the additional Event Tickets Settings related to the REST API.
+	 *
+	 * @since TBD
+	 *
+	 */
+	protected function hook_settings() {
+		add_filter( 'tribe_addons_tab_fields', array( tribe( 'tickets.rest-v1.settings' ), 'filter_tribe_addons_tab_fields' ) );
```

Comment:
We can use `tribe_callback( 'tickets.rest-v1.settings', 'filter_tribe_addons_tab_fields' )` to avoid fetching the instance unless needed.